lldb/tools/driver/Driver.cpp

```diff
@@ -821,6 +822,18 @@ int main(int argc, char const *argv[]) {
     }
   }
 
-  SBDebugger::Terminate();
+  // When terminating the debugger we have to wait on all the background tasks
+  // to complete, which can take a while. Print a message when this takes longer
+  // than 1 second.
+  {
+    std::future<void> future =
+        std::async(std::launch::async, []() { SBDebugger::Terminate(); });
```

Comment:
Is there any risk on calling Terminate() on a different thread (as opposed to printing the message on another thread and terminating in the main thread?

Reply:
I can't immediately think of a reason it wouldn't be safe to do, but I didn't look at every single terminator. Doing what you suggest is possible, though it would be slightly more convoluted and slightly less efficient. You'd have to spawn an async task and check periodically whether terminate has completed on the main thread. Something like:

```
bool terminated = false;
SBDebugger::Terminate();
terminated = true;
std::async(std::launch::async, [&terminated]() {
  for (size_t i = 0; i < 10; ++i) {
    if (terminated)
      return;
    std::this_thread::sleep_for(100ms);
   }
   fprintf(stderr, "Waiting for background tasks to complete...\n");
}
```

Let me know if you think that's worth the trade-off.

Reply:
If you think it's safe, it's fine with me.